src/sshagent/SSHAgent.cpp

```diff
@@ -217,8 +217,13 @@ void SSHAgent::databaseModeChanged(DatabaseWidget::Mode mode)
     } else if (mode == DatabaseWidget::ViewMode && !m_keys.contains(uuid.toHex())) {
         for (Entry* e : widget->database()->rootGroup()->entriesRecursive()) {
 
-            if (!e->attachments()->hasKey("KeeAgent.settings"))
+            if (e->group()->iconNumber() == Group::RecycleBinIconNumber) {
```

Comment:
This should be a check for `e->group() == db->metadata()->recycleBin()` (with a preceding check for `db->metadata()->db->recycleBinEnabled() && db->metadata()->recycleBin()`). Only checking the icon is not sufficient.